sklearn/metrics/classification.py

```diff
@@ -1056,8 +1193,10 @@ def _prf_divide(numerator, denominator, metric, modifier, average, warn_for):
     The metric, modifier and average arguments are used only for determining
     an appropriate warning.
     """
-    result = numerator / denominator
     mask = denominator == 0.0
+    denominator = denominator.copy()
```

Comment:
Why these changes?

Reply:
trying to remember. This silences a divide-by-zero warning, but I'm trying to remember why this did not come up with P/R/F.

Reply:
So the purpose is to silence the warning?
I think P/R/F manually filtered out these warnings with ``with np.errstate(divide='ignore', invalid='ignore')``, maybe we can do the same things here.

Reply:
Generally I prefer to avoid dividing by zero... using np.errstate is not threadsafe, and it's sensitive to context rather than clear from the code.

Reply:
If we change this, maybe we should remove ``np.errstate`` in P/R/F?
And this change is actually not backward compatible, I think we still need ``result[mask] = 0.0``. (though in P/R/F and jaccard, this won't make any difference since ``denominator = numerator + XXX``).

Reply:
When do we use this function with denominator < numerator?

Reply:
> When do we use this function with denominator < numerator?

Never, so it's not a big issue, but I guess we don't want things like ``_prf_divide(1, 0) = 1`` and ``_prf_divide(2, 0) = 2``?